src/unstructured_client/_hooks/custom/split_pdf_hook.py

```diff
@@ -439,3 +665,6 @@ def _clear_operation(self, operation_id: str) -> None:
         """
         self.coroutines_to_execute.pop(operation_id, None)
         self.api_successful_responses.pop(operation_id, None)
+        tempdir = self.tempdirs.pop(operation_id, None)
+        if tempdir:
```

Comment:
are there other areas we should consider cleaning up or does `_clear_operation` cover every case? curious if there are scenarios where if the api call fails, we don't clean up and leave files in the directory

Reply:
That's the limitation of the SDK... it's not supposed to fail cleaning on the condition that all the possible exceptions are caught (and it looks we catch them - but it's not a fireproof :D )
It would be much cleaner and resilient if we could use a nice context manager but because of the hooks we hack here we cannot.